### PR TITLE
Remove deprecated `InteractiveViewer.alignPanAxis`

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/integration_tests/flutter_gallery/lib/demo/shrine/supplemental/product_card.dart [hunk 1]
```diff
@@ -35,7 +35,7 @@ class ProductCard extends StatelessWidget {
 
     // The fontSize to use for computing the heuristic UI scaling factor.
     const double defaultFontSize = 14.0;
-    final double containerScaingFactor = MediaQuery.textScalerOf(context).scale(defaultFontSize) / defaultFontSize;
+    final double containerScalingFactor = MediaQuery.textScalerOf(context).scale(defaultFontSize) / defaultFontSize;
```
Comment (Piinks) [inline dev/integration_tests/flutter_gallery/lib/demo/shrine/supplemental/product_card.dart:38]: Looks like there are some unrelated text scaling changes

File: packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# file for instructions on testing these data driven fixes.
+
+# For documentation about this file format, see
+# https://dart.dev/go/data-driven-fixes.
+
+# * Fixes in this file are for InteractiveViewer from the widgets library. *
+version: 1
+transforms:
+  # Change made in https://github.com/flutter/flutter/pull/109014
+  - title: "Migrate to 'panAxis'"
```
Comment (Piinks) [inline packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml:22]: I think alignPanAxis = false => PanAxis.free and alignPanAxis = true => PanAxis.aligned.
Dart fix can do this, look for examples with `oneOf:`

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart [hunk 3]
```diff
@@ -319,19 +319,6 @@ class KeepAliveNotification extends Notification {
 /// consider using [AutomaticKeepAliveClientMixin], which uses a
 /// [KeepAliveHandle] internally.
 class KeepAliveHandle extends ChangeNotifier {
-  /// Trigger the listeners to indicate that the widget
-  /// no longer needs to be kept alive.
-  ///
-  /// This method does not call [dispose]. When the handle is not needed
-  /// anymore, it must be [dispose]d regardless of whether notifying listeners.
-  @Deprecated(
-    'Use dispose instead. '
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/automatic_keep_alive.dart:328]: If we've told users they should replace this with dispose, we should have a dart fix for it. The original author should be able to confirm. I think it was @dnfield?

**Review by Piinks — COMMENTED** [review]
Thanks for getting this ready! We have not announced any of these yet, so please do not land these until we have done so.

Can you separate the KeepAliveHandle and the InteractiveViewer parts into different PRs?
If there is a customer issue with one, it will end up affecting the other.

File: packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml (hunk 2 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml:22]: Right but `alignPanAxis` is already a no-op. Converting it to the new parameter changes the current behavior (see the PR description about the other dartfix).

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart (hunk 3 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:328]: KeepAliveHandle.release and KeepAliveHandle.dispose seem to have different semantics as the latter also releases all resources, and typically shouldn't be called a second time (see the example below).  The closer alternative seems to be `notifyListeners` but it's protected.

```dart
handler.release();
handler.dispose();
```
would be migrated to:
```dart
handler.dispose();
handler.dispose();
```
and the second dispose call is redundant and it's probably going to assert in debug mode.

File: dev/integration_tests/flutter_gallery/lib/demo/shrine/supplemental/product_card.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply dev/integration_tests/flutter_gallery/lib/demo/shrine/supplemental/product_card.dart:38]: Yeah I was trying to smuggle a typo fix. Will do that in a different PR then.

File: packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml (hunk 2 above)
Comment (Piinks) [reply packages/flutter/lib/fix_data/fix_widgets/fix_interactive_viewer.yaml:22]: It's unfortunate that we did not maintain backwards compatibility during the deprecation period and instead made a hard break going no-op, but the intention is for users to migrate to `PanAxis`. So we should absolutely provide a fix for this.

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart (hunk 3 above)
Comment (Piinks) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:328]: I think the motivation to deprecate was that folks were only calling release, and not calling dispose, but this one might be ok without a fix. We typically provide a fix in every case we can. Even if it is not a perfect fix, it helps users get to the final result.

**Review by Piinks — APPROVED** [review]
LGTM! Thanks for adding a fix!

**Pull request merged**